### PR TITLE
add FileBasedDatabase implementation

### PR DESCRIPTION
**Pull request by pswientek**

### CODE REVIEW

File: app/src/main/java/pl/futurecollars/invoicing/service/IdService.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private final Path idFilePath;
+    private final FileService fileService;
+
+    private int nextId = 1;
+
+    public IdService(Path idFilePath, FileService fileService) {
+        this.idFilePath = idFilePath;
+        this.fileService = fileService;
+
+        try {
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/service/IdService.java:16]: Try to move creating a file to FileService and make it generic

File: app/src/main/java/pl/futurecollars/invoicing/service/JsonService.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import pl.futurecollars.invoicing.model.Invoice;
+
+public class JsonService {
+
+    private final ObjectMapper objectMapper;
+
+    {
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/service/JsonService.java:16]: static block is fine, but in most case I don't like it - it is fine to just use no argument constructor

**Review by radoslawdabrowski — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/futurecollars/invoicing/service/IdService.java (hunk 1 above)
Comment (pswientek) [reply app/src/main/java/pl/futurecollars/invoicing/service/IdService.java:16]: I try to do it as we discussed yesterday. My changes did make several issues in the testing area, I got most of it covered but with the 4 remaining, I got stuck. Could you please aid me with it?

Comment (pswientek) [reply app/src/main/java/pl/futurecollars/invoicing/service/IdService.java:16]: nvm it. Got it done

**Review by radoslawdabrowski — APPROVED** [review]

**Pull request merged**